src/wazuh_modules/syscollector/syscollector_bsd.c

```diff
@@ -1303,6 +1312,133 @@ int getGatewayList(OSHash *gateway_list){
     return 0;
 }
 
+#ifdef __MACH__
+
+void sys_proc_mac(int queue_fd, const char* LOCATION){
+    char *timestamp;
+    time_t now;
+    struct tm localtm;
+    int random_id = os_random();
+
+    if (random_id < 0)
+        random_id = -random_id;
+
+    // Define time to sleep between messages sent
+    int usec = 1000000 / wm_max_eps;
+
+    now = time(NULL);
+    localtime_r(&now, &localtm);
+
+    os_calloc(TIME_LENGTH, sizeof(char), timestamp);
+
+    snprintf(timestamp,TIME_LENGTH-1,"%d/%02d/%02d %02d:%02d:%02d",
+            localtm.tm_year + 1900, localtm.tm_mon + 1,
+            localtm.tm_mday, localtm.tm_hour, localtm.tm_min, localtm.tm_sec);
+
+    pid_t * pids = calloc(0x1000, 1);
+    int count = proc_listallpids(pids, 0x1000);
```

Comment:
Why the limit of `0x1000` (4096) PIDs when reading the list of running processes? Are we sure it is enough to read all the processes? Is there any way to dynamically set that resource?